pallets/asset-index/src/lib.rs

```diff
@@ -1087,7 +1077,7 @@ pub mod pallet {
 			let index_token = Self::saft_equivalent(saft_nav)?;
 
 			// mint the given units of the SAFT asset into the treasury's account
-			T::Currency::deposit(asset_id, &Self::treasury_account(), units)?;
+			T::Currency::deposit(asset_id, &Self::treasury_account(), units)?; // ??? Bryan - what prevents this from minting other assets? e.g. DOT
```

Comment:
the question is, via which governance? If this is installed to Karura / Acala runtime, it is not acceptable to allow anyone other than root to have permission to mint arbitrary tokens

Reply:
I see. The thing with SAFT is, that there are no tokens (yet). [docs.polkadotindex.com/safts-within-the-pint-index](https://docs.polkadotindex.com/safts-within-the-pint-index)
So we would need some further restrictions on this, or disable the SAFT registry pallet for the Acala integration.
The latter would make more sense since `AssetId` here would be `CurrencyId` which is a fixed set right?

Reply:
One approach is do some sub currency id that is fully owned by PINT pallets. Then PINT pallets can have full privilege to this.
So we need: `type AssetId: From<PintAssetId> + TryInto<PintAssetId>` and 
```
enum PintAssetId {
  PINT,
  SAFT(u32), // or whatever identifier
}
```

and for Acala

```
enum CurrencyId {
 // current variants,
  Pint(PintAssetId),
}
```

Reply:
That's an elegant solution, we'll go with that.